lib/swaps/Swaps.ts

```diff
@@ -141,116 +132,152 @@ class Swaps extends EventEmitter {
       makerCurrency = baseCurrency;
     }
     const { takerAmount, makerAmount } = Swaps.calculateSwapAmounts(taker.quantity, maker.price);
+    const preimage = randomBytes(32);
+    const hash = createHash('sha256');
+    const r_hash = hash.update(preimage).digest('hex');
 
     const dealRequestBody: packets.DealRequestPacketBody = {
-      dealId,
       takerCurrency,
       makerCurrency,
+      r_hash,
       takerAmount,
       makerAmount,
-      takerPubKey: takerCurrency === 'BTC' ? this.pool.handshakeData.lndbtcPubKey! : this.pool.handshakeData.lndltcPubKey!,
       orderId: maker.id,
       pairId: maker.pairId,
       proposedQuantity: taker.quantity,
     };
 
     const deal: SwapDeal = {
       ...dealRequestBody,
-      myRole : SwapDealRole.Taker,
-      peerPubKey: this.pool.handshakeData.nodePubKey,
+      r_preimage: preimage.toString('hex'),
+      myRole: SwapDealRole.Taker,
       createTime: Date.now(),
     };
 
     this.addDeal(deal);
-    this.logger.debug('Initiating swap deal: ' + JSON.stringify(deal));
+    this.logger.debug('[SWAP] Initiating deal: ' + JSON.stringify(deal));
 
     const packet = new packets.DealRequestPacket(dealRequestBody);
 
     peer.sendPacket(packet);
-    return dealId;
+    return r_hash;
+  }
+
+  /**
+   * resolveHash resolve hash to preimage.
+   */
+  public resolveHash = async (args: { hash: string }) => {
```

Comment:
I don't think we need to keep the `args` parameter from the Service class. This can just take a single string parameter, and better to call it `r_hash` than `hash` I think to stay consistent.

Reply:
well, you may be right but resolveHash takes few parameters and not just the hash. 
Don't you find it wrong that we don't know about it? 
This has to do with the toObject discussion. 

I will take care of it when doing #431

Reply:
I'm not sure what you mean by other parameters? Which ones?

I'm just saying that the method signature should be `public resolveHash = async (hash: string)  => {`. There's no need to pass this method an object instead of just passing the value directly to be consistent with most other methods that accept only a handful of parameters.

Reply:
check the HashResolver proto definition:
```
service HashResolver {
    // ResolveHash is used by LND to request translation of Rhash to a pre-image.
    // the resolver may return the preimage and error indicating that there is no
    // such hash/deal
    rpc ResolveHash(ResolveRequest) returns (ResolveResponse) {}
}

message ResolveRequest {
    string hash = 1 [json_name = "hash"];
    uint32 timeout = 2 [json_name = "timeout"];
    uint32 height_now = 3 [json_name = "height_now"];
    int64  amount = 4 [json_name = "amount"];
}

message ResolveResponse {
    string preimage = 1 [json_name = "preimage"];
}
```
`ResolveRequest` has other fields in addition to the `hash`. But we don't know it.

It is really wrong.